src/hangar/dataset.py

```diff
@@ -128,7 +128,15 @@ def __len__(self):
         int
             number of samples the dataset contains
         '''
-        return len(self._Query.dataset_data_names(self._dsetn))
+        if not self._is_conman:
+            self._dataTxn = self._TxnRegister.begin_reader_txn(self._dataenv)
+        try:
+            DatasetRecordCountKey = parsing.dataset_record_count_db_key_from_raw_key(self._dsetn)
+            dset_len = int(self._dataTxn.get(DatasetRecordCountKey))
```

Comment:
The top level dataset method should make no assumptions about the structure of the records returened from a `get` call. Instead we send it to the appropriate methods in the `parsing` module which will decode and return the data which is expected.  Instead, the implementation should read like this:

```
try:
    dsetCountKey = parsing.dataset_record_count_db_key_from_raw_key(self._dsetn)
    dsetCountVal = self._dataTxn.get(dsetCountKey, default='0'.encode())
    dset_count = parsing.dataset_record_count_raw_val_from_db_val(dsetCountVal)
finally:
   ...
```

There's actually a bug in the straight up `casting` implementation due to the fact that the string is not decoded. This might have failed in some corner cases...

Also, it's not stated anywhere, but if you look through the `dataset` or `metadata` files, you'll see that the following convensions are generally objserved for variable names. 

```
    camelCase = raw (binary) encoded data which is used to `get` / `set` keys/values in the db
    underscore (ie. `foo_val`) = some native python type or data structure (typically used for inputs/outputs).
```

If you find any cases where this convention isn't observed, please open a PR! (There very well may be some procedures still around which were from initial prototype of this tool before it was pushed to github and I actually established a bit more standard conventions/style)